### PR TITLE
chore(gpu): clean overflowing sub memory

### PR DESCRIPTION
**Pull request by agnesLeroy**
closes: _please link all relevant issues_

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: backends/tfhe-cuda-backend/cuda/include/integer.h [hunk 1]
```diff
@@ -1157,7 +1157,7 @@ template <typename Torus> struct int_sc_prop_memory {
   }
 };
 
-template <typename Torus> struct int_single_borrow_prop_memory {
+template <typename Torus> struct int_overflowing_sub_memory {
```
Comment (bbarbakadze) [inline backends/tfhe-cuda-backend/cuda/include/integer.h:1160]: why  name it as `int_overflowing_sub_memory`, this structure  is for a function `unsigned_overflowing_propagate_subtraction_borrow`, old name is really not good, so we had to update it but maybe a different name to specify that it is for `unsigned_overflowing_propagate_subtraction_borrow`

File: backends/tfhe-cuda-backend/cuda/include/integer.h [hunk 2]
```diff
@@ -1169,10 +1169,10 @@ template <typename Torus> struct int_single_borrow_prop_memory {
 
   int_radix_params params;
 
-  int_single_borrow_prop_memory(cudaStream_t *streams, uint32_t *gpu_indexes,
-                                uint32_t gpu_count, int_radix_params params,
-                                uint32_t num_radix_blocks,
-                                bool allocate_gpu_memory) {
+  int_overflowing_sub_memory(cudaStream_t *streams, uint32_t *gpu_indexes,
```
Comment (bbarbakadze) [inline backends/tfhe-cuda-backend/cuda/include/integer.h:1172]: same about naming

File: backends/tfhe-cuda-backend/cuda/include/integer.h (hunk 1 above)
Comment (bbarbakadze) [reply backends/tfhe-cuda-backend/cuda/include/integer.h:1160]: also I don't know any other function that's using `unsigned_overflowing_propagate_subtraction_borrow` and can't imagine what else can use it, so maybe it`s fine to name it as `int_overflowing_sub_memory`

**Review by bbarbakadze — COMMENTED** [review]
Thank you, looks good to me, maybe think a bit more about naming.

File: backends/tfhe-cuda-backend/cuda/include/integer.h (hunk 1 above)
Comment (agnesLeroy) [reply backends/tfhe-cuda-backend/cuda/include/integer.h:1160]: Yeah this is really the only operation where this is used, so I know it's a bit inconsistent but it seemed easier to read like this to me.

**Review by bbarbakadze — APPROVED** [review]

**Pull request merged**